### DISCUSSION TITLE
Tkinter / Tk / Ttk used in Toga?

### DISCUSSION THREAD

**Discussion by White-Rabbit-Scientific** [beeware/toga#1652] [General]
Probably a silly question, and guessing the answer is "no" but I'll ask anyway... does Toga use Tkinter / Tk / Ttk at all, or is it entirely free of these libraries?

Does Toga use legacy UI libraries at all?

(Toga appears to be a fabulous piece of work, so kudos to the devs !!!)

**Comment by freakboy3742** [comment]
Toga does not use Tk in any way. It uses platform-native APIs to render GUIs. That means:
* Cocoa on macOS
* WinForms .Net on Windows
* GTK+ on Linux
* UIKit on iOS
* The android.* libraries on Android
* HTML/CSS for the Web

In that way, Toga is able to appear as close to a native app as possible. For example, there should be almost no look and feel difference between a Toga app on macOS, and an app written using Swift using Apple-provided tools; the same is true of every other supported platform.

**Reply by White-Rabbit-Scientific** [reply]
Awesome, thanks.
